docs/iris/src/further_topics/metadata.rst

```diff
@@ -0,0 +1,987 @@
+- the ``Conventions`` key has a **different value**, and
+- the ``grinning face`` key is **new**
+
+.. doctest:: metadata-combine
+
+    >>> attributes = {"Model scenario": "A1B", "Conventions": "CF-1.8", "grinning face": "🙂" }
+    >>> metadata = cube.metadata._replace(attributes=attributes)
+    >>> metadata != cube.metadata
+    True
+    >>> metadata.combine(cube.metadata).attributes
+    {'Model scenario': 'A1B'}
+
+The combined result for the ``attributes`` member only contains those
+**common keys** with **common values**.
+
+Note that, the ``combine`` method is **commutative**,
+
+.. doctest:: metadata-combine
+
+    >>> cube.metadata.combine(metadata) == metadata.combine(cube.metadata)
+    True
+
+Although, this is only the case when combining instances of the **same**
+metadata class. This is explored in a little further detail next.
+
+
+.. _combine like:
+
+Combine like with like
+""""""""""""""""""""""
+
+Akin to the :ref:`equal <metadata equality>` and
+:ref:`difference <metadata difference>` methods, only instances of **similar**
+metadata classes can be combined, otherwise a ``TypeError`` is raised,
+
+.. doctest:: metadata-combine
+
+    >>> cube.metadata.combine(longitude.metadata)
+    Traceback (most recent call last):
+    TypeError: Cannot combine 'CubeMetadata' with <class 'iris.common.metadata.DimCoordMetadata'>.
+
+Again, however, the :ref:`exception to the rule <exception rule>` also applies
+here i.e., support is provided between :class:`~iris.common.metadata.CoordMetadata` and
+:class:`~iris.common.metadata.DimCoordMetadata` metadata classes.
+
+For example, we can ``combine`` the metadata of the following
+:class:`~iris.coords.AuxCoord` and :class:`~iris.coords.DimCoord`,
+
+.. doctest:: metadata-combine
+
+    >>> forecast_period = cube.coord("forecast_period")
+    >>> longitude = cube.coord("longitude")
+
+First, let's see their associated metadata,
+
+.. doctest:: metadata-combine
+
+    >>> forecast_period.metadata
+    CoordMetadata(standard_name='forecast_period', long_name=None, var_name='forecast_period', units=Unit('hours'), attributes={}, coord_system=None, climatological=False)
+    >>> longitude.metadata
+    DimCoordMetadata(standard_name='longitude', long_name=None, var_name='longitude', units=Unit('degrees'), attributes={}, coord_system=GeogCS(6371229.0), climatological=False, circular=False)
+
+Before combining their metadata together,
+
+.. doctest:: metadata-combine
+
+    >>> forecast_period.metadata.combine(longitude.metadata)
+    CoordMetadata(standard_name=None, long_name=None, var_name=None, units=None, attributes={}, coord_system=None, climatological=False)
+    >>> longitude.metadata.combine(forecast_period.metadata)
+    DimCoordMetadata(standard_name=None, long_name=None, var_name=None, units=None, attributes={}, coord_system=None, climatological=False, circular=None)
+
+However, note that commutativity in this case cannot be honoured, for obvious reasons.
```

Comment:
Haha - the obvious reason is because the time can't have circular?  How have you managed to make it intelligently understand this!!  That's pretty cool given it isn't to do with the type of coord, but the scientific nature of the data being processed.